omniledger/service/service.go

```diff
@@ -669,32 +677,19 @@ func (s *Service) startPolling(scID skipchain.SkipBlockID, interval time.Duratio
 				// Pre-run transactions to look how many we can fit in the alloted time
 				// slot. Perhaps we can run this in parallel during the wait-phase?
 				log.Lvl3("Counting how many transactions fit in", interval/2)
-				var txsCollect ClientTransactions
-				cdbI := s.GetCollectionView(scID)
-				now := time.Now()
-				for len(txs) > 0 {
-					if s.verifyClientTx(scID, txs[0]) == nil {
-						var cin []Coin
-						for _, instr := range txs[0].Instructions {
-							_, cin, err = s.executeInstruction(cdbI, cin, instr)
-							if err != nil {
-								continue
-							}
-						}
-						if time.Now().Sub(now) < interval/2 {
-							txsCollect = append(txsCollect, txs[0])
-							txs = txs[1:]
-						} else {
-							log.Lvlf3("Got more transactions than what I can do in half the blockInterval. "+
-								"%d transactions left", len(txs))
-							break
-						}
-					} else {
-						log.Lvl3("Removing badly signed transaction")
-						txs = txs[1:]
-					}
+				cdb := s.getCollection(scID)
+				_, txOut, txBad, _, err := s.createStateChanges(cdb.coll, scID, txs, interval/2)
+				if err != nil {
+					log.Error("While choosing transactions, couldn't create state changes:", err)
+					continue
 				}
-				_, err = s.createNewBlock(scID, sb.Roster, txsCollect)
+
+				txs = txs[len(txOut)+len(txBad):]
```

Comment:
This is used to compute the unprocessed transactions, right? Maybe we can just have `createStateChanges` return that instead of `ctsBad`. I don't think `ctsBad` is used for anything else.